src/main/java/seedu/address/model/job/Job.java

```diff
@@ -32,7 +32,10 @@ public Job(String name, Money rate, float duration, boolean hasPaid, Set<Person>
         this.rate = rate;
         this.duration = duration;
         this.hasPaid = hasPaid;
-        this.persons.addAll(persons);
+
+        for (Person person: persons) {
+            this.persons.add(person.addJob(this));
+        }
```

Comment:
May result in duplicated jobs being added as Job is immutable.

Reply:
Thanks for the catch! I'm considering the pros and cons of various designs for how to manage a bidirectional relationship between the Job and Person classes efficiently. For now, though, the code has now been updated to remove jobs from the Person class so that the relationship is unidirectional. This should hopefully resolve any duplication issues.